File: backend/amoz/src/main/java/com/zpi/amoz/requests/AttributeCreateRequest.java
package com.zpi.amoz.requests;

import io.swagger.v3.oas.annotations.media.Schema;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;

import java.util.Optional;

@Schema(description = "Request do tworzenia nowego atrybutu, który zawiera nazwę atrybutu oraz wartość opcjonalną.")
public record AttributeCreateRequest(

        @Schema(description = "Nazwa atrybutu", example = "Kolor")
        @NotBlank(message = "Attribute name is required")
        @Size(max = 50, message = "Attribute name should not exceed 50 characters")
        String attributeName,

        @Schema(description = "Wartość atrybutu, jeśli istnieje", example = "Czerwony", nullable = true)
        Optional<@Size(max = 255) String> value

) {

    public AttributeCreateRequest(String attributeName, Optional<String> value) {
        this.attributeName = attributeName;
        this.value = value.isPresent() ? value : Optional.empty();
    }
}

File: frontend/app/src/main/java/com/example/amoz/requests/AttributeCreateRequest.kt
package com.example.amoz.requests

import kotlinx.serialization.Serializable
import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.Size

@Serializable
data class AttributeCreateRequest(

    @field:NotBlank(message = "Attribute name is required")
    @field:Size(max = 50, message = "Attribute name should not exceed 50 characters")
    val attributeName: String,

    @field:Size(max = 255, message = "Attribute value should not exceed 255 characters")
    val value: String? = null
)
